src/services/news.js

```diff
@@ -83,6 +83,37 @@ const getNotExpiredFormatted = async () => {
   return news;
 };
 
+/**
+ * Gets unexpired student news matching a search string
+ * @param {NewsItem[]} unexpiredNews the list of unfiltered news items
+ * @param {String} query the search query
+ * @return {Promise<any>} Student news
+ */
+const getFilteredNews = (unexpiredNews, query) => {
+  const filteredNews = [];
+  unexpiredNews.map((newsitem) => {
+    let queryparts = query.split(' ');
+    for (let i = 0; i < queryparts.length; i++) {
+      let querypart = queryparts[i];
+      if (querypart === '') {
+        continue;
+      }
```

Comment:
3 notes:

1. You forgot to lowercase `query`.
2. A `for of` loop is much more readable here.
3. `String.split()` doesn't return empty elements, so you don't need the empty string check here.

```suggestion
    for (let queryPart of query.toLowerCase().split(' ')) {
```

Reply:
So what about if the querypart = ''

Reply:
That doesn't happen because `split()` doesn't return empty strings.

Reply:
So it was returning an empty string earlier and that if case fixed it.

Reply:
It turns out that you're absolutely right. If the final character of `query` is a space, then split returns an empty string, which then becomes a query part and matches anything. This also happens if two spaces in a row appear in `query`.